ui/ts/pages/helpus.ts

```diff
@@ -0,0 +1,215 @@
+          // At the next tick, set the modal opacity to 1 and trigger a redraw so it will fade in.
+          setTimeout(
+            () => {
+              this.stage = Stage.Modal;
+              m.redraw();
+            },
+            0);
+        }
+
+        // Submit the modal data if the form is valid
+        modalSubmit(e: Event): void {
+          let target: HTMLButtonElement = <HTMLButtonElement>e.target;
+          if (target.form.checkValidity()) {
+
+            this.userData.save().then(() => {
+              // Change the modal size from the "Sign Up" to the "Thanks" size
+              this.stage = Stage.ThanksSize;
+              // After 400ms display the "Thanks" modal text
+              setTimeout(
+                () => {
+                  this.stage = Stage.ThanksText;
+                  m.redraw();
+                },
+                400);
+            });
+          }
+        }
+
+        // If the close button is hit, or the escape button is pressed, close the modal and remove "help-us" from the URL.
+        close(): void {
+          this.userData.attributes.dismissed = true;
+          this.userData.save().then(() => {
+            // Remove the event listener that listens for the escape key.
+            document.removeEventListener("keydown", this.closeOnEscapeBound);
+            let routeParts: string[] = m.route().split("?");
+            let route: string = routeParts[0];
+            let params: Object = m.route.parseQueryString(routeParts[1]);
+            delete params["help-us"];
+            // Fade the correct modal.
+            if (this.stage < Stage.ModalClose) {
+              this.stage = Stage.ModalClose;
+            } else {
+              this.stage = Stage.ThanksClose;
+            }
+            // Remove help-us from the URL after the fade completes.
+            setTimeout(
+              () => {
+                m.route(route, params);
+              },
+              200);
+          });
+        }
+
+        // Runs the close function when the escape key is pressed.
+        closeOnEscape(e: KeyboardEvent): void {
+          if (e.keyCode === 27) {
+            this.close();
+            m.redraw();
+          }
+        }
+      }
+
+      export function controller(): ModalController {
+        return new ModalController();
+      }
+
+      export function view(ctrl: ModalController): _mithril.MithrilVirtualElement {
+        let modalClass: string = "";
+        let containerClass: string = "";
+
+        if (ctrl.stage === Stage.ThanksSize) {
+          modalClass += ".thanks-pre";
+        }
+        if (ctrl.stage === Stage.ModalStart) {
+          modalClass += ".hide-modal";
+        }
+        if (ctrl.stage === Stage.ModalClose || ctrl.stage === Stage.ThanksClose) {
+          containerClass += ".close";
+        }
+
+        let close: () => void = ctrl.close.bind(ctrl);
+
+        if (ctrl.show) {
+          // Banner
+          if (ctrl.stage === Stage.Banner) {
```

Comment:
To be honest, I've always stayed away from switch fallthroughs because they're kind of error prone (as mentioned here in Javascript the Good Parts http://archive.oreilly.com/pub/a/javascript/excerpts/javascript-good-parts/bad-parts.html) but this is kind of the perfect place for it, so I guess I'll cave.